dash/development/_py_components_generation.py

```diff
@@ -1,16 +1,21 @@
 from collections import OrderedDict
 import copy
+import enum
+import numbers
 import os
+import typing
 from textwrap import fill, dedent
 
+from typing_extensions import TypedDict, NotRequired
 from dash.development.base_component import _explicitize_args
 from dash.exceptions import NonExistentEventException
 from ._all_keywords import python_keywords
 from ._collect_nodes import collect_nodes, filter_base_nodes
+from ._py_prop_typing import get_prop_typing, enums, shapes
 from .base_component import Component
 
 
-# pylint: disable=unused-argument,too-many-locals
+# pylint: disable=unused-argument,too-many-locals,too-many-branches
 def generate_class_string(
```

Comment:
if possible can we please change this to kw only arguments. Its difficult to see which one is which. 

```
def generate_class_string(*, typename, props, description, namespace, prop_reorder_exceptions=None, max_props=None). 
```